File: php/testcode/xss_021.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_xss_error_reflection
function xss_error_reflection(BenchmarkRequest $req): BenchmarkResponse {
    $page = $req->param('page');

    $html = '<html><body><p>Error: page "' . $page . '" not found.</p></body></html>'; // vuln-code-snippet vuln-line php_xss_error_reflection

    return BenchmarkResponse::html($html);
}
// vuln-code-snippet end php_xss_error_reflection

File: rust/testcode/xss_021.rs
//! CWE-79: User input interpolated directly into a JavaScript string literal in a script block.

// vuln-code-snippet start testcodeXss021
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let username = req.param("username");

    let html = format!("<script>var user = \"{}\";</script>", username); // vuln-code-snippet target-line testcodeXss021

    super::shared::BenchmarkResponse::ok(&html)
}
// vuln-code-snippet end testcodeXss021
